src/types/protocol.ts

```diff
@@ -159,15 +159,30 @@ export function isRosbridgeCallServiceMessage(
   return message.op === "call_service";
 }
 
-export interface RosbridgeServiceResponseMessage<TValues = unknown[]>
-  extends RosbridgeMessage {
+interface BaseRosbridgeServiceResponseMessage extends RosbridgeMessage {
   op: "service_response";
   id?: string;
   service: string;
-  values?: TValues;
   result: boolean;
 }
 
+/** If the service call failed, `values` will be a string error message. */
+interface FailedRosbridgeServiceResponseMessage
+  extends BaseRosbridgeServiceResponseMessage {
+  values?: string;
+  result: false;
+}
+
+interface SuccessfulRosbridgeServiceResponseMessage<TValues = undefined>
+  extends BaseRosbridgeServiceResponseMessage {
+  values: TValues;
+  result: true;
+}
+
+export type RosbridgeServiceResponseMessage<TValues = undefined> =
+  | FailedRosbridgeServiceResponseMessage
+  | SuccessfulRosbridgeServiceResponseMessage<TValues>;
```

Comment:
@bjsowa I've created a union type for `RosbridgeServiceResponseMessage` that enforces that the value returned is of type `TValues` if successful but is of type `string` if unsuccessful, does that sound right?